pkg/gitproviders/common.go

```diff
@@ -34,17 +42,41 @@ type GitProviderHandler interface {
 	CreatePullRequestToUserRepo(userRepRef gitprovider.UserRepositoryRef, targetBranch string, newBranch string, files []gitprovider.CommitFile, commitMessage string, prTitle string, prDescription string) (gitprovider.PullRequest, error)
 	CreatePullRequestToOrgRepo(orgRepRef gitprovider.OrgRepositoryRef, targetBranch string, newBranch string, files []gitprovider.CommitFile, commitMessage string, prTitle string, prDescription string) (gitprovider.PullRequest, error)
 	GetAccountType(owner string) (ProviderAccountType, error)
+	OauthConfig() OauthProviderConfig
+	GetUser(ctx context.Context, token *oauth2.Token) (*User, error)
+}
+
+// OauthProviderConfig is an abstraction around the oauth2.Config to allow for mocking.
+// This represents the subset of funcs the API server needs from the oauth2.Config.
+//counterfeiter:generate . OauthProviderConfig
+type OauthProviderConfig interface {
+	AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string
+	Exchange(ctx context.Context, code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error)
 }
 
 // making sure it implements the interface
 var _ GitProviderHandler = defaultGitProviderHandler{}
 
-func New() defaultGitProviderHandler {
-	return defaultGitProviderHandler{}
+type User struct {
+	Email string
+}
+
+// TODO: currently, this always returns the github implementation.
+// This eventually needs to return the different git provider implementations.
+func New(providerName ProviderName) (defaultGitProviderHandler, error) {
```

Comment:
Since this is exported and in a common file we should have a go doc comment. Same comment for the other exported functions